File: app/src/main/java/com/lxm/test/ui/SecondActivity.java
package com.lxm.test.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import com.github.mikephil.charting.charts.PieChart;
import com.lxm.test.L;
import com.lxm.test.R;
import com.lxm.test.base.BaseActivity;
import com.lxm.test.chart.PieChartManage;
import com.lxm.test.utils.ActivityManager;
import java.io.UnsupportedEncodingException;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;
import butterknife.BindView;
import butterknife.OnClick;

/**
  * @Copyright (C), 2020
  * @Author: ym
  * @Date:
  * @Description:
  */
public class SecondActivity extends BaseActivity {

    @BindView(R.id.btn_second)
    TextView btnSecond;
    @BindView(R.id.pie_chart)
    PieChart pieChart;
    private PieChartManage pieChartManage;
    private List<Float> list= new ArrayList<>();
    private List<String> sList=new ArrayList<>();
    private static final int total=100;
    private Float sum=0f;


    @Override
    public int getLayoutResource() {
        return R.layout.activity_second;
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        super.initView(savedInstanceState);
        btnSecond.setText(getString(R.string.btn_result));
        getRandomFloat();
        showPieChart(list);
    }

    private void getRandomFloat() {
        Random random=new Random();
        for (int i = 0; i <10; i++) {
            Float temp = random.nextFloat() * 10;
            list.add(i, temp);
            sList.add(i,
                    getRandomChar()+getRandomChar());
            sum=sum+ temp;
            if(sum>100){
               break;
            }
        }
    }
    /**
      * @Description: 生成汉字
      */
    private String getRandomChar() {
        String str = "";
        int heightPos;
        int lowPos;

        Random random = new Random();

        heightPos = (176 + Math.abs(random.nextInt(39)));
        lowPos = (161 + Math.abs(random.nextInt(93)));

        byte[] b = new byte[2];
        b[0] = (Integer.valueOf(heightPos)).byteValue();
        b[1] = (Integer.valueOf(lowPos)).byteValue();

        try {
            str = new String(b, "GBK");
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
            L.i("生成汉字错误");
        }

        return str;
    }

    private void showPieChart(final List<Float> list) {
        if (pieChartManage == null) {
            pieChartManage = new PieChartManage(pieChart);
        }
         pieChartManage.showPicChart
                ((ArrayList<Float>) list, (ArrayList<String>) sList, total);

    }

    @OnClick(R.id.btn_second)
    public void onClickView(View view) {
        switch (view.getId()){
            case R.id.btn_second:
                setResult(RESULT_OK, new Intent().putExtra("data",
                        ActivityManager.getInstance().currentActivity().getClass().getSimpleName()));
                finish();
                break;
        }
    }


}

File: app/src/main/java/com/lxm/test/ui/SecondActivity.kt
package com.lxm.test.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.github.mikephil.charting.charts.PieChart
import com.lxm.test.R
import com.lxm.test.base.BaseActivity
import java.util.ArrayList
import java.util.Random
import butterknife.BindView
import butterknife.OnClick
import com.lxm.test.NULL_CHARACTER
import com.lxm.test.chart.PieChartManage
import com.lxm.test.utils.ActivityManager
import com.lxm.test.utils.LogCompat.logI
import java.lang.Exception
import java.nio.charset.Charset
import kotlin.math.abs

/**
  * @Copyright (C), 2020
  * @Author: ym
  * @Date:
  * @Description:
  */
class SecondActivity : BaseActivity() {

    @BindView(R.id.btn_second)
    private var btnSecond:TextView ? = null
    @BindView(R.id.pie_chart)
    private var  pieChart : PieChart ? = null
    private var  pieChartManage : PieChartManage ? = null
    private var  list : MutableList<Float> = ArrayList()
    private var  sList : MutableList<String> = ArrayList()
    private val  total : Int = 100
    private var sum : Float = 0f


    override fun getLayoutResource(): Int {
        return R.layout.activity_second
    }

    override fun initView(savedInstanceState: Bundle?) {
        super.initView(savedInstanceState)
        btnSecond?.text = getString(R.string.btn_result)
        getRandomFloat();
        showPieChart(list);
    }

    private fun getRandomFloat() {
        val  random=Random()
        for (index in 0 until 10) {
            val temp = random.nextFloat() * 10
            list.add(index, temp)
            sList.add(index,
                    getRandomChar()+getRandomChar())
            sum += temp
            if(sum>100)
               break
        }
    }
    /**
      * @Description: 生成汉字
      */
    private fun getRandomChar() : String{

        var str = NULL_CHARACTER
        val random = Random()
        val heightPos = (176 + abs(random.nextInt(39)))
        val lowPos = (161 + abs(random.nextInt(93)))

        val b = ByteArray(2)
        b[0] = heightPos.toByte()
        b[1] = lowPos.toByte()

        try {
            str = String(b, Charset.forName("GBK"))
        } catch ( e : Exception) {
            e.printStackTrace()
            ("生成汉字错误").logI()
        }

        return str;
    }

    private fun showPieChart( list :MutableList<Float>) {
        if (pieChartManage == null)
            pieChartManage = PieChartManage(pieChart)
         pieChartManage?.showPicChart(list , sList , total)

    }

    @OnClick(R.id.btn_second)
    fun onClickView(view : View?) {
        when (view?.id){
             R.id.btn_second ->{
                 setResult(RESULT_OK,  Intent().putExtra("data",
                         ActivityManager.currentActivity?.javaClass?.simpleName))
                 finish()
             }
        }
    }


}